pkg/framework/plugins/pluginbuilder/pluginbuilder.go

```diff
@@ -47,28 +46,27 @@ func NewRegistry() Registry {
 	return Registry{}
 }
 
-func Register(name string, builderFunc PluginBuilder, exampleArg runtime.Object, balance bool, registry Registry) {
+func Register(name string, builderFunc PluginBuilder, exampleArg runtime.Object, registry Registry) {
```

Comment:
Plugin registration does not require passing its arguments given the same plugin builder can be invoked multiple times with different arguments.

Reply:
> Plugin registration does not require passing its arguments given the same plugin builder can be invoked multiple times with different arguments.

That's why I call them exampleArgs... https://github.com/kubernetes-sigs/descheduler/pull/1003#discussion_r1033529025

They are not meant to be "filled". This is just so when doing manual conversations latter we don't have to deal with raw runtime.Objects and reflect types and we know already what type of args each registered plugin holds. These will be always empty args as an example for comparison.

Ref from the other pr: https://github.com/knelasevero/descheduler/blob/7c4d2960ebeea39cf7ff81474cc4a99e3d2aa22d/pkg/api/v1alpha2/conversion.go#L95

Reply:
@ingvagabund this exampleArg thing is already making me avoid having to have a big switch case here as well: 				https://github.com/knelasevero/descheduler/blob/5edd5a7bd10276fba1ad077745aa5a1353737e67/pkg/descheduler/policyconfig.go#L167

It is just useful knowing the plugin to Arg type pair registered always, and being able to just pull it up from the registry. To me using the registry to store only New() funcs is a waste (And, again, this is what will make the manual conversion that I plan doing later possible, while no one figures out the native scheme conversion).

Reply:
:raised_hands: